### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by eypsrcnuygr**
#class_check - a method which checks for the available classes and apply my_select to them for later usage.
#my_each - a method which behaves like #each.
#my_select - a method which behaves like #select.
#my_all? - a method which behaves like #all?.
#my_any? - a method which behaves like #any?.
#my_none? - a method which behaves like #none?.
#my_count - a method which behaves like #count.
#my_map - a method which behaves like #map. This method was also modified to accept either a proc or a block.
#my_inject - which behaves like #inject. The first implementation throws a complexity error, so I've got the idea from @defoebrand and implement his idea while understanding the refactoring.
#multiply_els - which uses #my_inject to multiply elements accumulatively.

There was a partner issue that makes me wait nearly 2 weeks, so it's decided for me to work solo for this project.

### CODE REVIEW

**Comment by eypsrcnuygr** [comment]
Add more logic to my_all, my_none?, my_any? and my_map.
Hope this works!

File: bin/main.rb [hunk 1]
```diff
@@ -87,38 +93,34 @@ def my_count(arg = nil, &block)
...
-    arr.length.times { |x| collector.push(proc.call(arr[x])) }
+    if proc
+      arr.length.times { |x| collector.push(proc.call(arr[x])) }
+    elsif block_given?
+      arr.length.times { |x| collector.push(yield(x)) }
+    else
+      return to_enum
+    end
     collector
   end
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply bin/main.rb:108]:
```rb
2.6.5 :029 > array.my_map(&my_proc) == array.map(&my_proc)
 => false 
2.6.5 :030 > 
```

File: bin/main.rb [hunk 2]
```diff
@@ -50,33 +48,41 @@ def my_select(&block)
...
+      !class_check(arg).empty?
+    elsif block_given?
+      !arr.my_select(&block).to_a.empty?
+    elsif arr.my_select { |x| x == false || x.nil? }.empty?
+      true
     else
-      arr.my_select(&block).to_a.empty? ? false : true
+      false
     end
   end
```
[the remark this answers is not in the document]
Comment (eypsrcnuygr) [reply bin/main.rb]: I change the necessary line with my_all? and now it works properly.

File: bin/main.rb [hunk 3]
```diff
@@ -50,33 +48,41 @@ def my_select(&block)
...
       class_check(arg).empty?
+    elsif block_given?
+      arr.my_select(&block).to_a.empty?
+    elsif arr.my_select { |x| x == false || x.nil? }.empty?
+      false
     else
-      arr.my_select(&block).to_a.empty? ? true : false
+      true
     end
   end
```
[the remark this answers is not in the document]
Comment (eypsrcnuygr) [reply bin/main.rb]: I add the deleted line again now it works properly.

File: bin/main.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (eypsrcnuygr) [reply bin/main.rb]: I was iterating over and giving the index. So now I am giving the element instead of the index. Right now it works.

**Comment by eypsrcnuygr** [comment]
* Arrange my_any?, my_none?, and my_map.
* They were not behaving as the originals without code block and argument and now they work.
* my_map were giving the index instead of the element. Now it gives the element.

**Pull request merged**